### ISSUE TITLE
No autocomplete for Literal for __getitem__

### ISSUE THREAD

**Issue by kumaraditya303** [microsoft/pylance-release#1318]
## Environment data

-   Language Server version: v2021.5.3
-   OS and version: Windows 10 21H1
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.9.5

## Expected behaviour

It should show autocomplete for `__getitem__`

## Actual behaviour

No autocomplete for `__getitem__`

## Code Snippet / Additional information

![Screenshot (69)](https://user-images.githubusercontent.com/59607654/118935715-74f74980-b969-11eb-83f4-a869c60cccc0.png)

```python
# pyright: strict
from typing import Literal


class Foo:
    def __getitem__(self, key: Literal["chromium", "firefox", "webkit"]) -> int:
        return 1


f = Foo()
k = f[]
```

**Comment by jakebailey** [comment]
@bschnurr is this the bug you had noticed?

@kumaraditya303 If you switch your type checking mode up, do you see the literal completions working? I believe there's a bug there, but it'd be helpful to check.

**Comment by kumaraditya303** [comment]
@jakebailey Nope there isn't any autocomplete for literals

**Comment by erictraut** [comment]
This isn't a bug. It's a feature request.

It's technically possible for us to add completion suggestions for the case where a `__getitem__` method's `key` parameter is annotated with a union of literals, but it's a very specialized (and I think rare) case. I've never this done before. You may be the first person to do this.

The amount of code to support this is not huge, but it's also not trivial. We could leave this issue open to see if others upvote it.

**Comment by jakebailey** [comment]
Yes, I was confusing this with function argument literal completion, which does have a bug at the moment.

**Comment by kumaraditya303** [comment]
The pyright correctly identifies the type and also reports error if the value of key in not a matching literal, also if you want to support this then make sure this works for all the functions of the object protocol

**Comment by kumaraditya303** [comment]
They are plenty of use cases for this, the common one for `__getitem__` being creating a dict like object but there are many more sophisticated use cases for operator overloading

**Comment by bschnurr** [comment]
This issue has been fixed in version 2022.1.1, which we've just released. You can find the changelog here:  [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202211-12-january-2022)
